### PR TITLE
Port MLv1 `clearBreakouts` and `removeBreakout` to Clj

### PR DESCRIPTION
**Pull request by camsaul**
I was looking for an easy win to end my day and digging into MLv1 looking for things to remove and I realized some of the `Q_DEPRECATED` stuff related to breakouts could easily be ported to MLv2. So this is not a huge change but hopefully can serve as a blueprint for moving over more parts of MLv1

### CODE REVIEW

**Review by snoe — CHANGES_REQUESTED** [review]
https://github.com/metabase/metabase/pull/33605/files#diff-ae96f815f3ca33b3069c629a95fe050aa0d627ea6fc578d6cca7bcfb86647e3bR117 Looks like a front end lint caught an error

File: src/metabase/lib/breakout.cljc [hunk 1]
```diff
@@ -84,3 +86,42 @@
                  (cond-> col
                    pos (assoc :breakout-position pos))))
              (m/indexed cols))))))
+
+(mu/defn breakout-at-index :- [:maybe ::lib.schema.expression/expression]
+  "Return the breakout at `breakout-index` if there is one, otherwise return `nil`. Handles negative breakout
+  indecies."
```
Comment (snoe) [inline src/metabase/lib/breakout.cljc:92]: typo

File: src/metabase/lib/breakout.cljc [hunk 2]
```diff
@@ -84,3 +86,42 @@
...
+   (when-let [query-breakouts (breakouts query stage-number)]
+     (let [breakout-index (if (neg? breakout-index)
+                            (+ (count query-breakouts) breakout-index)
+                            breakout-index)]
+       (when (< -1  breakout-index (count query-breakouts))
+         (nth query-breakouts breakout-index))))))
+
+(mu/defn remove-breakout-at-index :- ::lib.schema/query
+  "If there is a breakout at `breakout-index`, return `query` with it removed; otherwise return `query` as-is. Handles
+  negative breakout indecies."
```
Comment (snoe) [inline src/metabase/lib/breakout.cljc:108]: typo

File: src/metabase/lib/breakout.cljc [hunk 3]
```diff
@@ -84,3 +86,42 @@
...
+     query)))
+
+(mu/defn clear-breakouts :- ::lib.schema/query
+  "Remove all breakouts from `query` at `stage-number`."
+  ([query]
+   (clear-breakouts query -1))
+
+  ([query        :- ::lib.schema/query
+    stage-number :- :int]
+   ;; TODO -- is this sufficient, or do we need to use `reduce` + `lib.remove-replace/remove-clause` here?
```
Comment (snoe) [inline src/metabase/lib/breakout.cljc:126]: I believe we should use lib.remove-replace here or we'll orphan references to the breakout.

File: src/metabase/lib/breakout.cljc [hunk 4]
```diff
@@ -84,3 +86,42 @@
...
+   (lib.util/update-query-stage query stage-number dissoc :breakout)))
```
Comment (metamben) [inline src/metabase/lib/breakout.cljc:127]: This can also invalidate other parts of the query.

**Review by metamben — CHANGES_REQUESTED** [review]
Tests with parts depending on the removed things are missing.

**Pull request closed without merging**